src/Common/src/Interop/Unix/libssl/Interop.libssl.cs

```diff
@@ -100,7 +100,10 @@ internal static partial class libssl
         [DllImport(Interop.Libraries.LibSsl)]
         internal static extern int BIO_ctrl_pending(IntPtr bio);
 
-        [DllImport(Interop.Libraries.LibSsl, EntryPoint = "SSL_CTX_set_quiet_shutdown")]
+        [DllImport(Interop.Libraries.LibSsl)]
         internal static extern void SSL_CTX_set_quiet_shutdown(IntPtr ssl, int mode);
+
+        [DllImport(Interop.Libraries.LibSsl, CharSet = CharSet.Ansi)]
+        internal static extern string SSL_get_version(IntPtr ssl);
```

Comment:
ssl_get_version returns a const char ptr. By having the DllImport defined to return string, the marshaller is going to try to free the returned ptr... that's a problem. This should instead be defined to return an IntPtr; you can then use the Marshal class to get the string.